src/ir/type-updating.cpp

```diff
@@ -306,18 +298,14 @@ Type GlobalTypeRewriter::getTempType(Type type) {
     return type;
   }
   if (type.isRef()) {
-    auto heapType = type.getHeapType();
-    if (auto it = typeIndices.find(heapType); it != typeIndices.end()) {
-      return typeBuilder.getTempRefType(typeBuilder[it->second],
-                                        type.getNullability());
+    auto ht = type.getHeapType();
+    if (auto it = typeIndices.find(ht); it != typeIndices.end()) {
+      ht = typeBuilder[it->second];
     }
-    // This type is not one that is eligible for optimizing. That is fine; just
-    // use it unmodified.
-    return type;
+    return typeBuilder.getTempRefType(ht, type.getNullability());
   }
```

Comment:
Did this code path change? Now it always calls `getTempRefType`, and I'm not sure why.

Reply:
It just simplifies the code a tiny bit to always create a temp type. I can undo this if you prefer; it has no bearing on correctness.

Reply:
Is it not slower to create a temp type?

Reply:
I guess it does do more work compared to just returning the original type. Will revert.